src/app/views/homepage/edit/EditHomepageController.jsx

```diff
@@ -0,0 +1,362 @@
+        }
+
+        if (saveHomepageChangesError) {
+            this.setState({ isSaving: false });
+            this.handleOnSaveError(`There was a problem saving your homepage changes`);
+        }
+
+        if (actions.isPreviewing) {
+            this.redirectTo(`/florence/collections/${this.props.params.collectionID}/homepage/preview`);
+        }
+
+        if (actions.isSubmittingForReview) {
+            const sendToReviewError = await this.sendToReview(this.props.params.collectionID, "/", formattedHomepageData);
+            this.setState({ isSaving: false });
+            notifications.add({
+                type: "positive",
+                message: `Homepage content saved!`,
+                isDismissable: true
+            });
+
+            if (sendToReviewError) {
+                this.setState({ isSaving: false });
+                this.handleOnSaveError(`There was a problem saving your homepage changes`);
+            } else {
+                this.redirectTo(`/florence/collections/${this.props.params.collectionID}`);
+            }
+        }
+    };
+
+    saveHomepageChanges = async (collectionID, homepageData) => {
+        await collections.savePageContent(collectionID, "/", homepageData).catch(error => {
+            log.event("Error saving homepage content", log.error(error));
+            console.error(`Error saving homepage content for '${this.props.params.collectionID}'`, error);
+            return error;
+        });
+    };
+
+    sendToReview = async (collectionID, homepageData) => {
+        await collections.setContentStatusToComplete(collectionID, "/", homepageData).catch(error => {
+            log.event("Error submitting for review", log.error(error));
+            console.error(`Error submitting for review '${this.props.params.collectionID}'`, error);
+            return error;
+        });
+    };
+
+    handleMarkAsReviewed = async () => {
+        try {
+            await collections.setPageContentAsReviewed(this.props.params.collectionID, "/");
+            this.redirectTo(`/florence/collections/${this.props.params.collectionID}`);
+        } catch (error) {
+            log.event(
+                "Error reviewing homepage content",
+                log.data({
+                    collectionID: this.props.params.collectionID
+                }),
+                log.error(error)
+            );
+            console.error(`Error reviewing content. Collection ID: '${this.props.params.collectionID}' for review. Error:`, error);
+            return error;
+        }
+    };
+
+    handleOnSaveError = message => {
+        notifications.add({
+            type: "warning",
+            message: `${message}. You can try again by pressing save.`,
+            isDismissable: true
+        });
+    };
+
+    redirectTo = route => {
+        window.location = window.location.origin + route;
```

Comment:
Is there a reason we're using `window.location` instead of router? e.g. `this.props.dispatch(push(uri))` ?

Reply:
Have made suggested change - tested and looks to still work as intended too